include/CLI/App.hpp

```diff
@@ -480,8 +446,18 @@ class App {
         Option *opt = add_option(option_name, fun, option_description, defaulted);
         opt->type_name(detail::type_name<T>())->type_size(-1);
 
+        opt->default_function([&variable]() {
```

Comment:
This would need to capture by value to make much sense, otherwise the default is going to change whenever the value is updated.

Reply:
It has to capture by reference, since it should only run if valid; however, it only gets "rendered" into a string once, when `->capture_default()` is run. This is a good point, though, if I add a the inheritable option, it needs to still capture at `->capture_default()` rather than when `help` is run.

Reply:
For example:

```cpp
int x;
app.add_option("--opt", x);
```

is allowed to capture a reference, but if you capture the value, you are reading from an uninitialized variable. Since when you capture by reference, the read from the variable only happens later when you run `->capture_default()`, there is no uninitialized variable read here.

Reply:
Is this primarily for help printing?  I guess I just assumed that it was also doing something with the default_str.  Or is that a completely separate mechanic and purpose.

Reply:
Like I said I hadn't looked much at the default stuff since I wasn't using it much.